### PR TITLE
bpo-46014: Add docs regarding `functools.singledispatch` changes

### PR DESCRIPTION
**Pull request by uriyyo**
#90172

https://bugs.python.org/issue46014

### CODE REVIEW

**Comment by iritkatriel** [comment]
https://bugs.python.org/issue46014 is closed. What is the status of this PR?

**Comment by AlexWaygood** [comment]
@iritkatriel that issue should perhaps be reopened. New functionality has been added for `singledispatch` in 3.11, but this functionality is not yet documented; this PR adds documentation for that functionality.

**Review by AlexWaygood — APPROVED** [review]
LGTM!

File: Doc/library/functools.rst [hunk 1]
```diff
@@ -436,6 +436,23 @@ The :mod:`functools` module defines the following functions:
      ...     for i, elem in enumerate(arg):
      ...         print(i, elem)
 
+   :data:`types.UnionType` and :data:`typing.Union` also can be used::
```
Comment (JelleZijlstra) [inline Doc/library/functools.rst]:
```suggestion
   :data:`types.UnionType` and :data:`typing.Union` can also be used::
```

File: Doc/whatsnew/3.11.rst [hunk 2]
```diff
@@ -245,6 +245,36 @@ fractions
   that an ``isinstance(some_fraction, typing.SupportsInt)`` check passes.
   (Contributed by Mark Dickinson in :issue:`44547`.)
 
+functools
+---------
+
+* :func:`functools.singledispatch` now supports :data:`types.UnionType`
+  and :data:`typing.Union` as dispatch argument.::
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.11.rst]:
```suggestion
  and :data:`typing.Union` as annotations to the dispatch argument.::
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Will merge this in a few days unless one of the other requested reviewers has more feedback.

**Pull request merged**